### ISSUE TITLE
Question: Could UNIQUE indexes be created at the end

### ISSUE THREAD

**Issue by urban-1** [dumblob/mysql2sqlite#88]
First of all, thanks for writing and maintain this - would have taken me eons to figure things out :)

Now, this is more of a question rather than an issue. I am converting a mysql db which dumps into 45G. The mysqldump part takes a few minutes while the import to sqlite takes 40-50 mins.

I have experimented with various parameters and PRAGMA but from what I read, the slowest operation here would be the index updates. Is it possible to move unique index creation at the end similarly to how keys are handled at the moment?

Note that this might be too much work for not much benefit (if I understand things right) due to the use of autoinc PKs which also create indexes.

**Comment by dumblob** [comment]
Generally I am not against the idea, but I would like to explore more the potential impact in the wild :wink:.

First thing which comes to my mind is whether there could exist a dump leveraging the speed of the indexes **during** its processing. In other words, a dump running complex queries right after the index got created.

I suppose this does not happen as mysqldump is rather a "dumb" tool, but still if you (or anyone else!) could check for such corner cases, it would make it much easier to proceed with your proposal :wink:. My time is in recent years very limited and I can not commit to this (at least not for now).

Thank you for understanding.

**Comment by urban-1** [comment]
Hi @dumblob, completely understand!

I experimented a bit with a hack: Removed the "UNIQUE" in a gsub before processing keys. This made the script process it as a key and append it at the end. In my case the performance benefit was negligible (if any). I suspect that autoinc indexes take more time than the uniques.

I agree, a lot more testing is needed - I am closing this. Maybe we reopen if someone else presents different results.

(Thanks for the fast response btw!)
